src/app/ReadClient.cpp

```diff
@@ -846,5 +894,17 @@ bool ReadClient::ResubscribeIfNeeded()
     return true;
 }
 
+void ReadClient::UpdateDataVersionFilters(const EndpointId aEndpointId, const ClusterId & aClusterId,
+                                          const DataVersion & aDataVersion)
+{
+    for (size_t index = 0; index < mReadPrepareParams.mDataVersionFilterParamsListSize; index++)
+    {
+        if (mReadPrepareParams.mpDataVersionFilterParamsList[index].mEndpointId == aEndpointId &&
```

Comment:
The documentation around SendAutoResubscribeRequest needs to make it clear that the data version filter params buffers also need to stay alive long enough _and_ that they will be written to.  Or something.  Because right now the contract around this stuff is entirely non-obvious.

Reply:
updated the documentation, thanks